doc/source/development/rfc/rfc85_policy_code_additions.rst

```diff
@@ -0,0 +1,88 @@
+contributors to the project.
+
+Motivation
+----------
+
+The GDAL project has historically been quite open to new code additions, including
+drivers that rely on proprietary licensed and/or closed-source SDKs (called
+"binary SDK" in the rest of this document). This approach is part
+of its strengths, but also comes at a price, with the software being larger
+and larger, with sometimes contributions being abandoned by their authors without
+new maintainers.
+
+Being a free & open-source project, the GDAL project will apply stronger scrutiny to
+code that require a binary SDK and encourage contributors to submit code additions
+that do not depend on such binary SDK or license such SDK under free & open-source
+license terms.
+
+Policy
+------
+
+- Drivers require a designated responsible contact, tracked in
+  https://github.com/OSGeo/gdal/wiki/Maintainers-per-sub-system
+
+- Contributions should follow other RFCs describing development rules, and come
+  with test scripts and sufficient documentation, covering usage and build instructions.
+
+- If the driver require a binary SDK not downloadable without cost, or that requires
+  a complicated registration process, the GDAL team is unlikely to support
+  driver inclusion.
+- GDAL supports multiple operating systems, and new drivers should support as many as
+  is practical. At a minimum, drivers should work with the latest and OS-vendor supported
+  releases of operating systems the driver is documented to work with.
+
+- If the binary SDK is no longer supported, or modernized to work with current
+  compilers and GDAL, the driver can be dropped. This rule also applies to open-source
+  dependencies that are no longer maintained or are superseded by other alternatives.
+
+- If the driver has unaddressed bugs, is breaking continuous integration (CI),
+  has caused CI bits to be disabled, or has not caught up to API modifications
+  requiring updates (extremely rare), the driver will be dropped from build
+  scripts, and thus will not be built without significant user intervention.
+
+- If critical/blocking issues reported in a driver are not addressed within a
+  2 month time-frame, it can be dropped from the tree entirely by a designated
+  GDAL maintainer for all releases going forward.
+
+- Contributors of significant code additions are expected to participate to the
+  day-to-day life of the project, and need to monitor closely the communication
+  channels of the project: issue tracker, mailing lists, etc. They are expected at
+  a minimum to respond in a timely manner to changes that affect the whole project:
+  CI, build scripts, upgrade of dependencies & build tools, documentation etc.
+  They are also expected to help sharing the burden of maintenance and participate
+  to the global health of the project by doing pull request reviews, tackling general
+  bug reports, functional enhancements, documentation and infrastructure enhancements,
+  etc.
```

Comment:
How about something like this?

```suggestion
- Contributors of significant code additions are expected to participate in the
  day-to-day life of the GDAL project, and need to monitor closely the communication
  channels of the project: issue tracker, mailing lists, etc.
  
- Maintainers are expected to supporting their contributions by triaging bug reports,
  reviewing related pull requests and RFCs, making functional enhancements, testing
  releases, and improving documentation, tests, and infrastructure.

- In addition, maintainers are expected to respond in a timely manner to wider 
  project changes (CI, build scripts, upgrade of dependencies, build tools,
  documentation, etc.) as it pertains to their contributions.
```

Reply:
> Maintainers are expected to supporting their contributions

this formulation seems to restrict the scope of their contribution only to their main area of interest. This is the minimum we can expect, but I believe we loose the "participate to the global health of the project" which is an important point. GDAL as a working project is not just the addition of juxtaposed drivers

Reply:
Sure, but if I came along and said "I've written a GDAL driver in my spare time for LibJPEG3000, I'm happy to maintain it in GDAL" should I be _expected_ to be fixing bugs in my spare time for drivers I have zero interest in or knowledge about? Of course, my help will always be welcome when I have time available, but "no, we won't accept the LibJPEG3000 driver unless you're committing to help maintain all of GDAL" doesn't seem like the message we want here?

Bearing in mind GDAL is now in a position where the wider project funds maintainers for a chunk of undirected maintenance, triage, core development, release-management, and other work. Obviously it hasn't been like that until very recently, and it still doesn't cover everything.

But maybe this RFC is more targeted at companies? In which case, very rough wording but how about:

```
- If the new contribution is commercial, the GDAL project expects the company will
  contribute either employee time, money, or both to the overall project health. This
  means either paying contributor(s) to work on improvements and maintenance
  for the wider project, or becoming a GDAL project sponsor.
```

Reply:
> But maybe this RFC is more targeted at companies?

It is targeted at (mostly) companies that wish to extend the reach of their binary SDKs through GDAL's very wide and valuable software distribution platform. There have been multiple cases where organizations have wanted to make a big "contribution" of code that benefits themselves and their communities without reciprocity of maintenance or responsibility to the GDAL community. 

This RFC is about laying down some of the rules of the road so that the PSC has the standing to toss out stuff that doesn't conform to the community's expectations. The expectations this RFC lays out, frankly, are not so challenging either – basically they have to take care of the thing and not simply provide a white elephant to the project that it must now either maintain or disappoint a customer/community by removing. 

The alternative, which is not desired because it is arbitrary and picks favorites, is to gate keep stuff on the way in. I think we want to give organizations a shot, but I also want to back up the PSC with some enforceability when it's clear that the entity's attention has waned or it or the community no longer has the resources to support the thing.

Reply:
> It is targeted at (mostly) companies that wish to extend the reach of their binary SDKs through GDAL's very wide and valuable software distribution platform

Let's spell that out clearly then: "_RFC 85: Policy regarding commercial drivers & binary SDKs_". The project can do whatever we want, we don't have to be "fair" or couch stuff in vague terms so it appears to apply to everyone. Holding companies to a higher standard of expectations than individuals is absolutely fine.

> This RFC is about laying down some of the rules of the road so that the PSC has the standing to toss out stuff that doesn't conform to the community's expectations.

The PSC _always_ has the standing to toss out stuff. If they decide to boot a driver because the company is absent, that's what they decide - there's no recourse/due-process or some appeal to a higher court. "Nobody here has have the enthusiasm to maintain it... bye! :wave:" I'm not saying that there shouldn't be clear/written expectations, but if they only apply to companies let's say so.

Reply:
> How about something like this?

Giving up some of my idealism, I'll take it as such. Thanks for helping shape this RFC.

Reply:
I quite liked my "you should be sponsoring the project" expectation from https://github.com/OSGeo/gdal/pull/5128#discussion_r786896901 😝 

(disclosure: work is a commercial GDAL sponsor)

Reply:
> I quite liked my "you should be sponsoring the project" expectation

I've no opposition to it. @hobu ?

Reply:
Something to the effect of this would do well: *RFC 85: PSC policy regarding binary SDKs*

Buying sponsorship is appreciated, but I don't think that has to be the expectation at all. The expectation is those who provided this thing, or their designates, need to answer the phone when issues come up. Otherwise, they have successfully externalized their user and software support costs onto the project. I think GDAL has been too lenient on this behavior in the past, and now that the project is so large, it needs to reevaluate.

> The PSC always has the standing to toss out stuff. 

Of course, but every act of doing so without some kind of policy that provides a rationale before the fact feels arbitrary. One person's junk and all that. The purpose of this RFC is to say, "yes, go ahead, but if you start externalizing costs onto the project, we can drop it without much notice." As you say, always true, but in this case much more explicit.

Reply:
> RFC 85: PSC policy regarding binary SDKs

For me, the scope of the RFC goes beyond binary SDKs. They are the typical example of the most problematic situations, but a large/complex driver, without any dependency or with open source dependencies, and whose maintainer would go away could also be a problem if the core team doesn't have the domain expertise